### PR TITLE
[Feature][MOD-181] Give warning on moderation detail page if reviewer input not submitted

### PR DESCRIPTION
**Pull request by sheriefvt**
Follow-up PR to rebase a previous PR #104 (will be closed) on future-release/0.4.0 

## Problem
If a reviewer accidentally exits the decision form we don't want them to navigate away from the page without making a decision.

## Goal
Give warning on moderation detail page if reviewer input (e.g. new/edit comment or change of decision) not submitted and reviewer tries to leave the page.

## Changes

- Update preprint-status-banner component
- Update preprint-detail page template, controller, and router
- Update translation file
- Add new addon ember-onbeforeunload for case 1 (see next)
- Add new unit tests

**Two cases**
- When user reloads or closes the page or navigate to OSF or another external app

- When user navigates to another route with-in the reviews app

## Ticket 
https://openscience.atlassian.net/browse/MOD-181

### CODE REVIEW

File: app/components/preprint-status-banner/component.js
Comment (aaxelb) [inline app/components/preprint-status-banner/component.js]: This doesn't use `submission.reviewsState`

File: app/components/preprint-status-banner/component.js
Comment (aaxelb) [inline app/components/preprint-status-banner/component.js]: For rejected submissions, the "default" is `REJECTED`, so this doesn't prompt the user if they try to leave after changing from rejected to accepted. It might be good to save an `initialDecision` somewhere, which is the initial selected value.

File: app/components/preprint-status-banner/template.hbs
Comment (aaxelb) [inline app/components/preprint-status-banner/template.hbs]: I think we want this `reviewActions` still

File: app/templates/preprints/provider/preprint-detail.hbs
Comment (aaxelb) [inline app/templates/preprints/provider/preprint-detail.hbs]: It looks like `preprint-status-banner` doesn't actually use `userHasEnteredReview`, it just sets it to notify the controller of a change. Passing an action here would be a better/cleaner fit for that than a two-way binding. The component could use that action instead of `closeReviewerFeedback`.

**Review by aaxelb — CHANGES_REQUESTED** [review]
Had a few small suggestions, and found a little bug.

File: app/components/preprint-status-banner/component.js
Comment (aaxelb) [inline app/components/preprint-status-banner/component.js]: Is this property necessary? With `initialDecision` it seems like `decisionChanged`/`userActivity` are all you need.

File: app/controllers/preprints/provider/preprint-detail.js
Comment (aaxelb) [inline app/controllers/preprints/provider/preprint-detail.js]: This name seems misleading -- it doesn't close anything.

File: app/components/preprint-status-banner/component.js
Comment (sheriefvt) [reply app/components/preprint-status-banner/component.js]: There is a special case where submission status is pending, however, user decision is 'accept' on the feedback form. I used the pendingStateNotChanged property to check for it.

Comment (aaxelb) [reply app/components/preprint-status-banner/component.js]: In that case, `initialDecision` and `decision` would both be `ACCEPTED`, so `decisionChanged` would be false, which is correct. Do we have to treat that case specially?

Comment (sheriefvt) [reply app/components/preprint-status-banner/component.js]: Remember our discussion regarding the case where a user might change her/his decision (for pending preprints), then receives the warning, if the user restored decision to accept --> still should continue recieve the warning.

File: app/controllers/preprints/provider/preprint-detail.js
Comment (sheriefvt) [reply app/controllers/preprints/provider/preprint-detail.js]: what about 'setUserEnteredReview' ?

Comment (aaxelb) [reply app/controllers/preprints/provider/preprint-detail.js]: Sure, your call. As long as the name describes what the thing is or what the thing does.

File: app/components/preprint-status-banner/component.js
Comment (aaxelb) [reply app/components/preprint-status-banner/component.js]: Ah, right. That doesn't need to be a special case, though -- you could set `userActivity = true` whenever the user clicks on a decision or enters text in the comment box. That might let you simplify some of the logic in this component, which is good because it looks like it's impossible to accept a pending preprint:

noaccept

File: app/components/preprint-status-banner/component.js
Comment (aaxelb) [inline app/components/preprint-status-banner/component.js]: It's confusing to have both `decisionUpdated` and `decisionChanged`, try to use more distinct names.

File: app/components/preprint-status-banner/component.js
Comment (aaxelb) [inline app/components/preprint-status-banner/component.js]: `pendingStateNotChanged` doesn't exist anymore.

File: app/components/preprint-status-banner/component.js
Comment (aaxelb) [inline app/components/preprint-status-banner/component.js]: Could this be simplified to `or('commentEdited', 'decisionValueToggled')`, getting rid of the intermediate `decisionUpdated`?

File: app/components/preprint-status-banner/component.js
Comment (aaxelb) [inline app/components/preprint-status-banner/component.js]: It looks like `initialDecision` is now just the same as `submission.reviewsState`, could we use that instead?

File: app/components/preprint-status-banner/template.hbs
Comment (aaxelb) [inline app/components/preprint-status-banner/template.hbs]: `closeFeedback` doesn't do much -- could this be `onHide=(action setUserEnteredReview userActivity)`?

File: app/templates/preprints/provider/preprint-detail.hbs
Comment (aaxelb) [inline app/templates/preprints/provider/preprint-detail.hbs]: You could use the [`mut` helper](https://emberjs.com/api/ember/2.16/classes/Ember.Templates.helpers) here, like `setUserEnteredReview=(action (mut userHasEnteredReview))`, which would let you delete `setUserEnteredReview`.

File: app/templates/preprints/provider/preprint-detail.hbs
Comment (aaxelb) [inline app/templates/preprints/provider/preprint-detail.hbs]: Minor: `onClick=(action modal.close)` would be better, so there's only one path out of the modal.

File: app/templates/preprints/provider/preprint-detail.hbs
Comment (aaxelb) [inline app/templates/preprints/provider/preprint-detail.hbs]: This could be `onHide=(action (mut showWarning false))`, which would let you delete `hideWarning`.

**Review by aaxelb — COMMENTED** [review]
The functionality looks good! I just had some small suggestions for reducing the amount of code.

File: app/controllers/preprints/provider/preprint-detail.js
Comment (aaxelb) [inline app/controllers/preprints/provider/preprint-detail.js]: It doesn't look like `hideWarning` is used anymore.

**Review by aaxelb — COMMENTED** [review]
Just one more small thing and this looks good to go 👍

File: app/routes/preprints/provider/preprint-detail.js
Comment (aaxelb) [inline app/routes/preprints/provider/preprint-detail.js]: Looks like `currentUser` isn't used anywhere.

**Review by aaxelb — DISMISSED** [review]
Code is nice and clean, and works perfectly!

Just found one little unused injection, otherwise 💯

**Review by laurenbarker — APPROVED** [review]
Thanks for removing the unused injection! I'll merge once tests pass.

**Pull request merged**
